### PR TITLE
Handle missed slot correctly

### PR DESCRIPTION
**Pull request by F4ever**

### CODE REVIEW

**Comment by vgorkavenko** [comment]
It looks like we should handle the same for `/validators` request

File: app/beacon.py [hunk 1]
```diff
@@ -70,11 +78,26 @@ def __init__(self, url, slots_per_epoch):
...
+
         try:
             return int(response.json()['data']['message']['body']['execution_payload']['block_number'])
         except KeyError as error:
             logging.error(f'Response [{response.status_code}] with text: {str(response.text)} was returned.')
-            raise KeyError from error
+            raise error
+
+    def get_slot_for_report(self, ref_slot: int, epochs_per_frame: int, slots_per_epoch: int):
+        for slot_num in range(ref_slot, ref_slot - epochs_per_frame * slots_per_epoch + 1, -1):
```
Comment (avsetsin) [inline app/beacon.py:92]:
```suggestion
        for slot_num in range(ref_slot, ref_slot - epochs_per_frame * slots_per_epoch - 1, -1):
```
as it's a decreasing range

Comment (F4ever) [reply app/beacon.py]: - 1 would be ours, previous report

**Review by avsetsin — APPROVED** [review]
lgtm

File: app/beacon.py (hunk 1 above)
Comment (F4ever) [reply app/beacon.py]: Removed +1 at all

**Review by vgorkavenko — APPROVED** [review]

**Review by infloop — APPROVED** [review]

**Pull request merged**